tests/subsys/net/lib/nrf_cloud/nrf_cloud_test/src/nrf_cloud_test.c

```diff
@@ -0,0 +1,654 @@
+ * return with permission denied error (-EACCES) when not idle
+ */
+ZTEST(nrf_cloud_test, test_init_not_idle)
+{
+	struct nrf_cloud_init_param params = {
+		.event_handler = event_handler,
+		.client_id = NULL,
+	};
+
+	zassert_equal(STATE_IDLE, nfsm_get_current_state(),
+		"nrf_cloud lib should be in the idle state (uninitialized) at the start of test");
+
+	/* Call nrf_cloud_init twice with the first one
+	 * being successful and check if the second one
+	 * will fail since the state will not be idle
+	 */
+	nct_init_fake.custom_fake = fake_nct_init__succeeds;
+	nfsm_init_fake.custom_fake = fake_nfsm_init__succeeds;
+	nrf_cloud_codec_init_fake.custom_fake = fake_nrf_cloud_codec_init__succeeds;
+	nrf_cloud_init(&params);
+
+	int ret = nrf_cloud_init(&params);
+
+	zassert_not_equal(STATE_IDLE, nfsm_get_current_state(),
+		"nrf_cloud lib should not be in the idle state after initialization");
+
+	zassert_equal(-EACCES, ret, "return should be -EACCES when not idle");
+}
+
+/* Verify that nrf_cloud_init function will throw an error
+ * of invalid argument (-EINVAL) when given null params
+ */
+ZTEST(nrf_cloud_test, test_init_null_params)
+{
+	zassert_equal(STATE_IDLE, nfsm_get_current_state(),
+		"nrf_cloud lib should be in the idle state (uninitialized) at the start of test");
+
+	int ret = nrf_cloud_init(NULL);
+
+	zassert_equal(-EINVAL, ret, "return should be -EINVAL with NULL params");
+}
+
+/* Verify that nrf_cloud_init function will throw an
+ * error of invalid argument (-EINVAL) when a null
+ * event handler is passed in as part of its parameters
+ */
+ZTEST(nrf_cloud_test, test_init_null_handler)
+{
+	struct nrf_cloud_init_param params = {
+		.event_handler = NULL,
+		.client_id = NULL,
+	};
+
+	zassert_equal(STATE_IDLE, nfsm_get_current_state(),
+		"nrf_cloud lib should be in the idle state (uninitialized) at the start of test");
+
+	int ret = nrf_cloud_init(&params);
+
+	zassert_equal(-EINVAL, ret, "return should be -EINVAL with NULL event handler");
+}
+
+/* Verify that nrf_cloud_init function succeeds given valid internal
+ * state, event handler with full modem FOTA update is DISABLED.
+ * Other tests will not need FMFU since it is already tested here.
+ */
+ZTEST(nrf_cloud_test, test_init_success_no_fota)
+{
+	struct nrf_cloud_init_param params = {
+		.event_handler = event_handler,
+		.client_id = NULL,
+	};
+
+	zassert_equal(STATE_IDLE, nfsm_get_current_state(),
+		"nrf_cloud lib should be in the idle state (uninitialized) at the start of test");
+
+	/* Expect all fakes to run successfully */
+	nct_init_fake.custom_fake = fake_nct_init__succeeds;
+	nfsm_init_fake.custom_fake = fake_nfsm_init__succeeds;
+	nrf_cloud_codec_init_fake.custom_fake = fake_nrf_cloud_codec_init__succeeds;
+
+	int ret = nrf_cloud_init(&params);
+
+	zassert_ok(ret, "return should be 0 when init success");
+}
+
+/* Verify that nrf_cloud_init function succeeds given valid internal
+ * state, event handler with full modem FOTA update is ENABLED.
```

Comment:
"given valid internal state, event handler with" is confusing to me -- can you reword this?

You write the same thing in ZTEST(nrf_cloud_test, test_init_success_no_fota) as well.